tests/components/airos/test_config_flow.py

```diff
@@ -402,3 +469,239 @@ async def test_reconfigure_unique_id_mismatch(
+        MOCK_DISC_DEV2[MAC_ADDRESS]: MOCK_DISC_DEV2,
+    }
+
+    flow_start = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    menu = await hass.config_entries.flow.async_configure(
+        flow_start["flow_id"], {"next_step_id": "discovery"}
+    )
+
+    discover_flow = await hass.config_entries.flow.async_configure(menu["flow_id"])
+
+    assert discover_flow["type"] is FlowResultType.FORM
+    assert discover_flow["step_id"] == "select_device"
+
+    expected_options = {
+        MOCK_DISC_DEV1[MAC_ADDRESS]: (
+            f"{MOCK_DISC_DEV1[HOSTNAME]} ({MOCK_DISC_DEV1[IP_ADDRESS]})"
+        ),
+        MOCK_DISC_DEV2[MAC_ADDRESS]: (
+            f"{MOCK_DISC_DEV2[HOSTNAME]} ({MOCK_DISC_DEV2[IP_ADDRESS]})"
+        ),
+    }
+    actual_options = (
+        discover_flow["data_schema"].schema[vol.Required(MAC_ADDRESS)].container
+    )
+    assert actual_options == expected_options
+
+    # Select one of the devices
+    select_flow = await hass.config_entries.flow.async_configure(
+        discover_flow["flow_id"], {MAC_ADDRESS: MOCK_DISC_DEV1[MAC_ADDRESS]}
+    )
+
+    assert select_flow["type"] is FlowResultType.FORM
+    assert select_flow["step_id"] == "configure_device"
+    assert (
+        select_flow["description_placeholders"]["device_name"]
+        == MOCK_DISC_DEV1[HOSTNAME]
+    )
+
+    # Provide credentials and complete the flow
+    mock_airos_client.status.return_value.derived.mac = MOCK_DISC_DEV1[MAC_ADDRESS]
+    mock_airos_client.status.return_value.host.hostname = MOCK_DISC_DEV1[HOSTNAME]
+
+    credentials_flow = await hass.config_entries.flow.async_configure(
+        select_flow["flow_id"],
+        {
+            CONF_USERNAME: DEFAULT_USERNAME,
+            CONF_PASSWORD: "test-password",
+            SECTION_ADVANCED_SETTINGS: MOCK_ADVANCED_SETTINGS,
+        },
+    )
+
+    assert credentials_flow["type"] is FlowResultType.CREATE_ENTRY
+    assert credentials_flow["title"] == MOCK_DISC_DEV1[HOSTNAME]
+    assert credentials_flow["data"][CONF_HOST] == MOCK_DISC_DEV1[IP_ADDRESS]
+
+
+async def test_discover_flow_with_existing_device(
+    hass: HomeAssistant, mock_discovery_method, mock_airos_client
+) -> None:
+    """Test that discovery ignores devices that are already configured."""
+    # Add a mock config entry for an existing device
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=MOCK_DISC_EXISTS[MAC_ADDRESS],
+        data=MOCK_CONFIG,
+    )
+    mock_entry.add_to_hass(hass)
+
+    # Mock discovery to find both a new device and the existing one
+    mock_discovery_method.return_value = {
+        MOCK_DISC_DEV1[MAC_ADDRESS]: MOCK_DISC_DEV1,
+        MOCK_DISC_EXISTS[MAC_ADDRESS]: MOCK_DISC_EXISTS,
+    }
+
+    flow_start = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    menu = await hass.config_entries.flow.async_configure(
```

Comment:
can we reuse `result` everywhere?

Reply:
We surely can but I'd suggest we do that in a follow-up PR as to not change code outside of the PRs scope? If we want it 'now' I can change the newer parts only, but it makes more sense to do a single change to 'result, result2, result3' everywhere.